ws2812b/src/ws2812b_lld.c

```diff
@@ -1,363 +1,394 @@
-        "nop                                 \n"// H t2
-        "nop                                 \n"// H t3
-        "nop                                 \n"// H t4
-        "nop                                 \n"// H t5
-        "nop                                 \n"// H t6
-        RGBLedPin_ReSet                         // L t1
-        "nop                                 \n"// L t1
-        "nop                                 \n"// L t2
-        "nop                                 \n"// L t3
-        "nop                                 \n"// L t4
-        "nop                                 \n"// L t5
-        "nop                                 \n"// L t6
-                                                // L DECW 2, JREQ 1, 2 = 5 =>   t14
-//--------------------------------------------------------
-//--------------------------------------------------------
-        "DECW    Y                           \n"//2
-        "JREQ    L:lb_exit                   \n"//1 on loop, 2 jmp to exit
-        "JP      L:lb_begin_loop             \n"//5
-        "lb_exit:nop");
-  
-    __set_interrupt_state(int_state);
-}
\ No newline at end of file
+/** 
+  ******************************************************************************
+  * @file    ws2812b_lld.c
+  * @brief   Low Level Driver for WS2812B Type RGB/RGBW LED strips
+  * @author  https://github.com/j3qq4hch
+  * @author  https://github.com/mnemocron
+  * @date    2020-10-27
+  * 
+  * @note    IAR Embedded Workbench 3.11.1
+  *          shared components 8.3.2.5988
+  * @note    tested with RGBW LED type: IN-PI55QATPRPGPBPW-XX
+  * 
+  ******************************************************************************
+  */
+
+/**@todo is this include file necessary? it compiles without it */
+// #include <iostm8s103f3.h>
+#include <intrinsics.h>
+#include "ws2812b_lld.h"
+
+// depending on RGBW or RGB type it is 3 or 4 bytes to manage
+#ifdef WS2812B_MODE_RGBW
+  #define COL_PER_LED (4)
+
+  const RGBColor_t RED            = {255,   0,   0,   0};
+  const RGBColor_t GREEN          = {  0, 255,   0,   0};
+  const RGBColor_t BLUE           = {  0,   0, 255,   0};
+  const RGBColor_t YELLOW         = {255, 255,   0,   0};
+  const RGBColor_t MAGENTA        = {255,   0, 255,   0};
+  const RGBColor_t BLACK          = {  0,   0,   0,   0};
+  const RGBColor_t WHITE          = {255, 255, 255,   0};
+#endif
+
+#ifdef WS2812B_MODE_RGB
+  #define COL_PER_LED (3)
+
+  const RGBColor_t RED            = {255,   0,   0};
+  const RGBColor_t GREEN          = {  0, 255,   0};
+  const RGBColor_t BLUE           = {  0,   0, 255};
+  const RGBColor_t YELLOW         = {255, 255,   0};
+  const RGBColor_t MAGENTA        = {255,   0, 255};
+  const RGBColor_t BLACK          = {  0,   0,   0};
+  const RGBColor_t WHITE          = {255, 255, 255};
+#endif
+
+unsigned char LedsArray[NB_LEDS * COL_PER_LED];
+unsigned int nbLedsBytes = NB_LEDS * COL_PER_LED;
+extern unsigned char NodeId;

```

Comment:
it's better to use sized types - uint8_t and uint16_t in this case. `int` and `char` is almost obsolete in modern C coding, they are used when size doesn't matter. Not in this case when the data is shared with ASM code